capella_git_hooks/fix_links.py

```
@@ -1,113 +1,242 @@
+            if element.text == "":
+                element.getparent().remove(element)
+                continue
+
+            targetfrag = element.text.rsplit("/", 1)[-1]
+            frags = [i for i in model.trees if i.name == targetfrag]
+            if len(frags) != 1:
+                raise RuntimeError(
+                    f"Ambiguous fragment name: {element.text}"
+                    " - please report this as bug at"
+                    " https://github.com/DSD-DBS/capella-git-hooks"
                 )
 
-            elem.attrib[key] = value
+            sourcefrag = model.find_fragment(element)
+            if sourcefrag == frags[0]:
+                element.getparent().remove(element)
+                continue
 
-    if changed:
-        exs.write(tree, path, line_length=line_length, siblings=True)
+            link = os.path.relpath(frags[0], sourcefrag.parent)
+            if link != element.text:
+                element.text = link
+                dirty = True
 
-    return changed
+        for attr, value in element.attrib.items():
+            if attr == "href":
+                include_target_type = False
+            else:
+                include_target_type = None
 
+            try:
+                links = tuple(helpers.split_links(value))
+            except ValueError:
+                continue
 
-def commit_changes(changed_files_to_be_committed: list[str]):
-    """Add and commit the provided list of files using a predefined message."""
-    subprocess.call(["git", "add"] + changed_files_to_be_committed)
-    subprocess.call(
-        ["git", "commit", "-m", "fix[by-script]: merge tool index-prefix"]
-    )
+            new_links: list[str] = []
+            for link in links:
+                _, _, target_id = link.partition("#")
+                try:
+                    target = model.follow_link(None, target_id)
+                except KeyError:
+                    LOGGER.error("Cannot repair dangling link: #%s", target_id)
```

The user will never find this error message. Maybe we should think about collecting error messages and let the hook fail after repairing the model on best effort basis?